snowfakery/parse_recipe_yaml.py

```diff
@@ -190,12 +190,19 @@ def parse_structured_value(name: str, field: Dict, context: ParseContext) -> Def
         namespace, name = function_name.split(".")
         plugin = context.parser_macros_plugins.get(namespace)
     if plugin:
-        func = getattr(plugin, name)
-        rc = func(context, args)
-        return rc
-    else:
-        args = parse_structured_value_args(args, context)
-        return StructuredValue(function_name, args, **context.line_num(field))
+        try:
+            func = getattr(plugin, name)
+            rc = func(context, args)
+            return rc
+        except AttributeError as e:
```

Comment:
Are you sure you want to handle AttributeError while running the function? That could be quite confusing if the plugin func raises an AttributeError. I suspect you only want the try/except around the getattr.